tests/models/swin/test_modeling_tf_swin.py

```diff
@@ -0,0 +1,363 @@
+            hidden_states = outputs.hidden_states
+
+            expected_num_layers = getattr(
+                self.model_tester, "expected_num_hidden_layers", len(self.model_tester.depths) + 1
+            )
+            self.assertEqual(len(hidden_states), expected_num_layers)
+
+            # Swin has a different seq_length
+            image_size = to_2tuple(self.model_tester.image_size)
+            patch_size = to_2tuple(self.model_tester.patch_size)
+
+            num_patches = (image_size[1] // patch_size[1]) * (image_size[0] // patch_size[0])
+
+            self.assertListEqual(
+                list(hidden_states[0].shape[-2:]),
+                [num_patches, self.model_tester.embed_dim],
+            )
+
+            reshaped_hidden_states = outputs.reshaped_hidden_states
+            self.assertEqual(len(reshaped_hidden_states), expected_num_layers)
+
+            batch_size, num_channels, height, width = reshaped_hidden_states[0].shape
+
+            reshaped_hidden_states = tf.reshape(reshaped_hidden_states[0], (batch_size, num_channels, height * width))
+            reshaped_hidden_states = tf.transpose(reshaped_hidden_states, (0, 2, 1))
+
+            self.assertListEqual(
+                list(reshaped_hidden_states.shape[-2:]),
+                [num_patches, self.model_tester.embed_dim],
+            )
+
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            inputs_dict["output_hidden_states"] = True
+            check_hidden_states_output(inputs_dict, config, model_class)
+
+            # check that output_hidden_states also work using config
+            del inputs_dict["output_hidden_states"]
+            config.output_hidden_states = True
+
+            check_hidden_states_output(inputs_dict, config, model_class)
+
+    def test_for_image_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_image_classification(*config_and_inputs)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in TF_SWIN_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = TFSwinModel.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+
+@require_vision
+@require_tf
+class TFSwinModelIntegrationTest(unittest.TestCase):
+    @cached_property
+    def default_feature_extractor(self):
+        return (
+            AutoFeatureExtractor.from_pretrained("microsoft/swin-tiny-patch4-window7-224")
+            if is_vision_available()
+            else None
+        )
+
+    @slow
+    def test_inference_image_classification_head(self):
+        model = TFSwinForImageClassification.from_pretrained(
+            "microsoft/swin-tiny-patch4-window7-224",
+            # FIXME - remove once model has been reviewed and weights pushed to the hub
+            from_pt=True,
+        )
```

Comment:
Suggestion be be added once model architecture approved and weights pushed to the hub

```suggestion
        model = TFSwinForImageClassification.from_pretrained(
            "microsoft/swin-tiny-patch4-window7-224",
        )
```

Reply:
Would suggest one line since it fits ;-)
```suggestion
        model = TFSwinForImageClassification.from_pretrained("microsoft/swin-tiny-patch4-window7-224")
```